src/plone/restapi/deserializer/dxcontent.py

```diff
@@ -78,7 +81,20 @@ def __call__(self, validate_all=False, data=None):  # noqa: ignore=C901
                             'message': e.doc(), 'field': name, 'error': e})
                     else:
                         field_data[name] = value
-                        if value != dm.get():
+                        dm_value = dm.get()
+
+                        # This is required in case that we can compare
+                        # offset-naive and offset aware objects. We convert all
+                        # offset-naive datetimes to UTC timezone first and then
+                        # compare them
+                        if isinstance(value, datetime) and \
+                           isinstance(dm_value, datetime):
+                            if value.tzinfo is None:
+                                value = value.replace(tzinfo=pytz.UTC)
+                            if dm_value.tzinfo is None:
+                                dm_value = dm_value.replace(tzinfo=pytz.UTC)
```

Comment:
Wouldn't it be better to localize the datetime values to the portal's timezone?

Something like:
```
from plone.app.event.base import default_timezone
dtz = default_timezone(self.context, as_tzinfo=True)
dtz.localize(value)
```

Or if you have to avoid a dependency on plone.app.event here:
```
value = DateTime(
                value.year(),
                value.month(),
                value.day(),
                value.hour(),
                value.minute()
).asdatetime()
```
That uses the default timezone of DateTime, which is the servers timezone (which should actually be set to UTC but often is set in the local zone).